libs/design-system/src/lib/DatePicker/DatePicker.tsx

```diff
@@ -25,11 +25,41 @@ export interface DatePickerProps {
     maxCalendarDate?: string
     popperPlacement?: PopperJs.Placement
     popperStrategy?: PopperJs.PositioningStrategy
+    dateFormat?: string
 }
 
-function toFormattedStr(date: string | null) {
+function toFormattedStr(date: string | null, dateFormat: string): string {
     if (!date) return ''
-    return DateTime.fromISO(date).toFormat(INPUT_DATE_FORMAT)
+    return DateTime.fromISO(date).toFormat(dateFormat)
+}
+
+// function to get mask from the given format
+function getMaskArray(dateFormat: String): string[] {
+    return dateFormat
+        .split('/')
+        .map((keyword) => keyword.trim())
+        .join('')
+        .split('')
+        .map((keyword) => keyword.toUpperCase())
+}
+
+function getDateFormatPattern(dateFormat: string): string {
+    const dateComponents = dateFormat.split('/')
+    const pattern = dateComponents
+        .map((component) => {
+            if (component.includes('d')) {
+                return ' ## ' // Placeholder for day
+            } else if (component.includes('m') || component.includes('M')) {
```

Comment:
This is looking great, thanks for taking this on! Just a few small questions:

- Do we need to check both upper and lower case here? If so, should we do it for day and year as well?
- In the getMaskArray could you use a regex to remove slashes and spaces in one go, then upper case and split to array?

Reply:
Concerning the initial item, we only need to address "M," but I will adjust all of them to perform case-insensitive checks for added safety. As for the second point, I will manage the modifications accordingly.

Reply:
I've addressed the comments. Kindly review the pull request and inform me if any further modifications are necessary.